### PR TITLE
fixes #24 - failing child processes

### LINKED ISSUE #24: failing child process, pass null to done
I don't think this is correct.

example

```js
asyncDone(function () {
  return cp.spawn('not_existing_command', ['hello'])
}, function (err, res) {
  console.log(err, res)
  // err always `null`
})
```
it always pass the result of spawn to res, no matter it executes `echo hello` or not exising command.

Because this `onSuccess` in [index.js#L43](https://github.com/phated/async-done/blob/master/index.js#L43)

### PR DESCRIPTION
**Pull request by tunnckoCore**
#24 
- add tests for it
- fix streams tests

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ function asyncDone(fn, cb){
   function done(){
     d.removeListener('error', onError);
     d.exit();
-    return cb.apply(null, arguments);
+    return cb.apply(cb, arguments);
```
Comment (phated) [inline index.js]: why is callback being applied as the context?  it doesn't seem that would effect anything.

**Comment by phated** [comment]
Just 1 comment, everything else looks good

File: index.js (hunk 1 above)
Comment (tunnckoCore) [inline index.js]: Hm, right.

**Comment by phated** [comment]
released as 1.1.1. thanks!

**Comment by tunnckoCore** [comment]
I also thanks! :sparkles:

**Pull request merged**